google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITE2ETracingTest.java

```diff
@@ -0,0 +1,557 @@
+
+  private static final int TRACE_FORCE_FLUSH_MILLIS = 5000;
+
+  private static final int TRACE_PROVIDER_SHUTDOWN_MILLIS = 1000;
+
+  private static Key KEY1;
+
+  // Random int generator for trace ID and span ID
+  private static Random random;
+
+  private static TraceExporter traceExporter;
+
+  // Required for reading back traces from Cloud Trace for validation
+  private static TraceServiceClient traceClient_v1;
+
+  // Custom SpanContext for each test, required for TraceID injection
+  private static SpanContext customSpanContext;
+
+  // Trace read back from Cloud Trace using traceClient_v1 for verification
+  private static Trace retrievedTrace;
+
+  private static String rootSpanName;
+  private static Tracer tracer;
+
+  // Required to set custom-root span
+  private static OpenTelemetrySdk openTelemetrySdk;
+
+  private static String projectId;
+
+  private static DatastoreOptions options;
+
+  private static Datastore datastore;
+
+  @TestParameter boolean useGlobalOpenTelemetrySDK;
+
+  @TestParameter({"default", "test-db"})
+  String datastoreNamedDatabase;
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    projectId = DatastoreOptions.getDefaultProjectId();
+    logger.info("projectId:" + projectId);
+
+    // TODO(jimit) Make it re-usable w/ InMemorySpanExporter
+    traceExporter =
+        TraceExporter.createWithConfiguration(
+            TraceConfiguration.builder().setProjectId(projectId).build());
+
+    traceClient_v1 = TraceServiceClient.create();
+
+    random = new Random();
+  }
+
+  @Before
+  public void before() throws Exception {
+    // Set up OTel SDK
+    Resource resource =
+        Resource.getDefault().merge(Resource.builder().put(SERVICE_NAME, "Sparky").build());
+
+    if (isUsingGlobalOpenTelemetrySDK()) {
+      openTelemetrySdk =
+          OpenTelemetrySdk.builder()
+              .setTracerProvider(
+                  SdkTracerProvider.builder()
+                      .setResource(resource)
+                      .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                      .setSampler(Sampler.alwaysOn())
+                      .build())
+              .buildAndRegisterGlobal();
+    } else {
+      openTelemetrySdk =
+          OpenTelemetrySdk.builder()
+              .setTracerProvider(
+                  SdkTracerProvider.builder()
+                      .setResource(resource)
+                      .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                      .setSampler(Sampler.alwaysOn())
+                      .build())
+              .build();
+    }
+
+    // Initialize the Datastore DB w/ the OTel SDK. Ideally we'd do this is the @BeforeAll method
+    // but because gRPC traces need to be deterministically force-flushed, datastore.shutdown()
+    // must be called in @After for each test.
+    DatastoreOptions.Builder optionsBuilder;
+    if (isUsingGlobalOpenTelemetrySDK()) {
+      optionsBuilder =
```

Comment:
I think a lot of this can be added to https://github.com/googleapis/java-datastore/blob/main/google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/RemoteDatastoreHelper.java for shared use.

Reply:
Added setting of OTel options to RemoteDatatstoreHelper. Now, any test using the RemoteDatastoreHelper can enable OTel by simply providing the SDK object.